spec/compiler/codegen/and_spec.cr

```diff
@@ -18,11 +18,11 @@ describe "Code gen: and" do
   end
 
   it "codegens and with bool and int 1" do
-    run("struct Bool; def to_i; 0; end; end; (false && 2).to_i").to_i.should eq(0)
+    run("struct Bool; def to_i!; 0; end; end; (false && 2).to_i!").to_i.should eq(0)
```

Comment:
Is this rename really needed? It's just a spec, I don't think it breaks anything if we use `to_i`, right?

Reply:
Oh, nevermind, it does because we later invoke it on `2`.

Reply:
Yeah, ideally we should be able to use exceptions with the JIT runner that does not include prelude. But that is a pending thing to be done. But the specs test more the dispatch and typing than the actual behavior.